File: src/Types/Elemento.hs
module Types.Elemento (
    Elemento(..),
    prioridadeElemento
) where

data Elemento = Fogo | Agua | Neve deriving (Eq, Read)

instance Show Elemento where
    show Fogo = "FOGO"
    show Agua = "ÁGUA"
    show Neve = "NEVE"

prioridadeElemento :: Elemento -> Elemento -> Bool
prioridadeElemento Fogo Neve = True
prioridadeElemento Neve Agua = True
prioridadeElemento Agua Fogo = True
prioridadeElemento _ _ = False

File: Prolog/src/Types/elemento.pl
:- module(elemento, [elemento/1,
                     prioridade_elemento/2,
                     mostrar_elemento/2]).

elemento(fogo).
elemento(agua).
elemento(neve).

prioridade_elemento(fogo, neve).
prioridade_elemento(neve, agua).
prioridade_elemento(agua, fogo).
prioridade_elemento(_, _):- false.

mostrar_elemento(fogo, 'FOGO', [fg(red)]).
mostrar_elemento(agua, 'ÁGUA', [fg(blue)]).
mostrar_elemento(neve, 'NEVE', [fg(white)]).


